### PR TITLE
Fixed header for DataTable

### PR DESCRIPTION
**Pull request by quadreex**

### CODE REVIEW

File: src/components/Form/CheckboxRaw.js [hunk 1]
```diff
@@ -117,6 +124,7 @@ CheckboxRaw.propTypes = {
    * adds required attribute to the checkbox
    */
   required: PropTypes.bool,
+  wrapperClassName: PropTypes.string,
```
Comment (fspoettel) [inline src/components/Form/CheckboxRaw.js]: Please add doc (even if it might seem clear what this does)

File: src/components/DataTable/DataTable.js [hunk 2]
```diff
@@ -162,14 +164,16 @@ class DataTable extends Component {
   }
 
   renderHead() {
+    const { fixedHeader } = this.props;
     const { columns, id, sortBy, sortDirection } = this.state;
 
     return (
       <thead>
-        <tr>
+        <tr className={fixedHeader ? 'slds-text-title_caps' : null}>
```
Comment (fspoettel) [inline src/components/DataTable/DataTable.js]: Why does `fixedHeader` imply text in caps?

File: src/components/DataTable/DataTable.js [hunk 3]
```diff
@@ -7,6 +7,8 @@ import { propTypes as tablePropTypes } from '../Table/Table';
 
 import defaultRowRenderer from './defaultRowRenderer';
 
+const FULL_HEIGHT_STYLE = { height: '100%' };
```
Comment (fspoettel) [inline src/components/DataTable/DataTable.js]: Please add an example to Storybook

File: src/components/DataTable/defaultSelectAllRenderer.js [hunk 4]
```diff
@@ -1,9 +1,16 @@
...
 
 import { CheckboxRaw } from '../../';
 
-export default ({ allSelected, dataKey, onSelectAll, tableId }) => { // eslint-disable-line react/prop-types
+const FIXED_STYLE = { marginLeft: '-0.25rem' };
+
+export default ({ allSelected, dataKey, fixed, onSelectAll, tableId, truncate }) => { // eslint-disable-line react/prop-types
   const checkboxId = `${tableId}-${dataKey}`;
+  const wrapperClassName = cx([
+    { 'slds-truncate': truncate },
```
Comment (fspoettel) [inline src/components/DataTable/defaultSelectAllRenderer.js]: What is being truncated here? As far as I am aware, this only renders a checkbox and no text

File: src/components/DataTable/defaultSelectAllRenderer.js [hunk 5]
```diff
@@ -12,14 +19,17 @@ export default ({ allSelected, dataKey, onSelectAll, tableId }) => { // eslint-d
...
-      />
+      <div className={wrapperClassName} style={fixed ? FIXED_STYLE : null}>
+        <CheckboxRaw
+          checked={allSelected}
+          hideLabel
+          id={checkboxId}
+          label="Select All"
+          name="options"
+          onChange={onSelectAll}
+          wrapperClassName={fixed ? 'slds-th__action' : null}
```
Comment (fspoettel) [inline src/components/DataTable/defaultSelectAllRenderer.js]: Hard to tell why this change is necessary without an example

File: src/components/DataTable/DataTable.js (hunk 2 above)
Comment (quadreex) [reply src/components/DataTable/DataTable.js]: Just took the markup from the SF docs for a fixed header. Maybe we can remove this.

File: src/components/Form/CheckboxRaw.js (hunk 1 above)
Comment (quadreex) [reply src/components/Form/CheckboxRaw.js]: added
